blocks/api/raw-handling/special-comment-converter.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Browser dependencies
+ */
+const { COMMENT_NODE } = window.Node;
+
+/**
+ * Looks for `<!--more-->` comments, as well as the `<!--more Some text-->`
+ * variant and its `<!--noteaser-->` companion, and replaces them with a custom
+ * element representing a future block.
+ *
+ * The custom element is a way to bypass the rest of the `raw-handling`
+ * transforms, which would eliminate other kinds of node with which to carry
+ * `<!--more-->`'s data: nodes with `data` attributes, empty paragraphs, etc.
+ *
+ * The custom element is then expected to be recognized by any registered
+ * block's `raw` transform.
+ *
+ * @param {Node} node The node to be processed.
+ * @return {void}
+ */
+export default function( node ) {
+	if (
+		node.nodeType !== COMMENT_NODE ||
+		node.nodeValue.indexOf( 'more' ) !== 0
+	) {
+		// We don't specificially look for `noteaser`, meaning that if one is
+		// found on its own (and not adjacent to `more`), it will be lost.
+		return;
+	}
+
+	// Grab any custom text in the comment
+	const customText = node.nodeValue.slice( 4 ).trim();
```

Comment:
Amusing side note: this fixes a bug in the `more` handling currently in the parser, which requires a space between "more" and the custom text. Core doesn't require that space. 🙂

Reply:
Oh, I missed the previous discussion wherein @dmsnell learned yet another wonderful weird thing about Core. 🙃